corehq/util/timer.py

```diff
@@ -224,3 +224,23 @@ def _inner(self, *args, **kwargs):
                 return fn(self, *args, **kwargs)
         return _inner
     return decorator
+
+
+DURATION_REPORTING_THRESHOLD = "_duration_reporting_threshold"
+
+
+def set_request_duration_reporting_threshold(seconds):
+    """Decorator to override the default reporting threshold for a view.
+
+    If requests to the view take longer than the threshold a Sentry event
+    will get created.
+
+    :param seconds: Requests that take longer than this many seconds to process
+        will be reported to Sentry. See ``corehq.middleware.LogLongRequestMiddleware``
+        for where the duration check takes place.
+    """
+    def decorator(view):
+        setattr(view, DURATION_REPORTING_THRESHOLD, seconds)
```

Comment:
I had imagined, but didn't specify explicitly (sorry), that this would create a timedelta
```suggestion
        threshold = timedelta(seconds=seconds)
        setattr(view, DURATION_REPORTING_THRESHOLD, threshold)
```